fvm/src/kernel/default.rs

```diff
@@ -729,7 +733,7 @@ where
     // TODO(M2) merge new_actor_address and create_actor into a single syscall.
     fn new_actor_address(&mut self) -> Result<Address> {
         let oa = self
-            .resolve_to_key_addr(&self.call_manager.origin(), false)
```

Comment:
Conformance tests are failing here because before we _already_ had a key address here, so we didn't query the state tree to resolve it from the ID address. Now, we are forced to query the state tree, which is also a small performance penalty. Let's track both the original sender address and the sender ID in the Call Manager.

The more correct solution would be to introduce an `ActorPointer` type that could be initialised from an ID or from another addresses, and would transparently handle and memoise resolutions.

Reply:
That would also unify the mess of Actor ID, ID addresses, and non-ID addresses.

Reply:
> Now, we are forced to query the state tree, and that part is not available in the test vector.

I'm confused. We resolve the origin in the Executor, so that _shouldn't_ be causing this issue.

Reply:
Oh, I see. It's the other way around.